src/Workspaces/Core/Portable/Options/GlobalOptionService.cs

```diff
@@ -270,7 +270,7 @@ public SerializableOptionSet GetSerializableOptionsSnapshot(ImmutableHashSet<str
             var changedOptionsKeysSerializable = _changedOptionKeys
                 .Where(key => serializableOptions.Contains(key.Option) && (!key.Option.IsPerLanguage || languages.Contains(key.Language!)))
                 .ToImmutableHashSet();
-            return new SerializableOptionSet(languages, optionService, serializableOptions, serializableOptionValues, changedOptionsKeysSerializable);
+            return new SerializableOptionSet(optionService, serializableOptionValues, changedOptionsKeysSerializable);
         }
 
         private ImmutableDictionary<OptionKey, object?> GetSerializableOptionValues(ImmutableHashSet<IOption> optionKeys, ImmutableHashSet<string> languages)
```

Comment:
@mavasani this method (And the one above) are the critical ones.  Above you can see we get the list of serializable options.  then below we *force* retrieval of all the values for them (none are done lazily afaict).  so the list of options matches the list of keys in the dict we pass in.